helpers/authentication.js

```diff
@@ -82,17 +82,22 @@ const populateCurrentUser = (req, res) => {
 };
 
 const checkConsent = (req, res) => {
-    if (res.locals.currentRole == "Lehrer" || res.locals.currentRole == "Administrator") {
+    if (res.locals.currentRole === "Lehrer" || res.locals.currentRole === "Administrator") {
         return Promise.resolve();
     }
     else {
         return api(req).get('/consents/', {qs: { userId: res.locals.currentUser._id }})
             .then(consents => {
-                const consent = consents.data[0];
-                if (consent.redirect == '/dashboard/' || req.baseUrl == "/firstLogin") {
-                    return Promise.resolve();
+                if (consents.data[0]){
+                    const consent = consents.data[0];
+                    if (consent.redirect === '/dashboard/' || req.baseUrl === "/firstLogin") {
+                        return Promise.resolve();
+                    } else {
+                        res.redirect(consent.redirect);
+                        return Promise.reject("firstLogin was not completed, redirecting...");
```

Comment:
@Langleu gibt es eine schönere Art an dieser Stelle einen redirect zu setzen und den Prozess nicht weiter fortzusetzen?
Im moment wird es als unhandled rejection in die console geschrieben...

Reply:
Fällt mir auf die schnelle nichts ein, aber besser als "Can't set Headers after they've been sent"